kerdock.yml

```diff
@@ -0,0 +1,51 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+# code id, physical, logical are all lower case
+code_id: kerdock
+physical: bits
+logical: bits
+
+name: 'kerdock'
+introduced: '\url{https://doi.org/10.1016/S0019-9958(72)90376-2}'
+
+description: 'Kerdock code is a \([[2^m, 2^{2m}, 2^{m-1} - 2^{frac{m}{2}-1}]]\) binary code where \(m\) is even. The size of code book is \(|2^{m-1}||R(1,m)|\) where \(R(1,m)\)
+is the corresponding first order Reed Muller code. The code is non linear and can exist for perimeters for which there is no linear code (See Nordstrom-Robinson). The relative
+minimum distance tends to \(\frac{1}{2}\) as \(m\) becomes large.'
+
+#protection: ''
+
+features:
+
+  rate: 'The transmission rate is \(\frac{2m}{2^m}\) which tends to 0 as \(m\) becomes large, hence these codes are asymptotically poor.'
+
+  encoders:
+    - 'Boolean functions are of the form \(Q(X) + l(X) + b\)' where \(Q(X) is \sum_{1 \leq i < j \leq m} a_{ij}X_{i}X_{j}\), \(l(X)\) is linear in \(m\) variables and \(b\) is 
+       in \(\mathbb{F}_{2}\). 
+    - 'Codewords can be considered as evaluations of these functions over \(\mathbb{F}_{2}^{m}\).'
+    - 'The matrix \(A = [a_{ij}]\) constitute the Kerdock set \(K\) (See notes).' 
+
+  decoders:
+    - 'Soft decision decoding involves extending the Fast Hadamard Transform decoding algorithm for the binary first-order Reed-Muller code to Kerdock code.'
+    - 'Complexity of soft decision decoding algorithm: \(4^m\) multiplications and \(m4^m\) additions \cite{arXiv:math/0207208}.'
+    
+
+realizations:
+  - 'Kerdock codes find application in digital fingerprinting.'
+  - 't - collision secure schemes can be designed to detect a pirated copy once t users have colluded \cite{doi:10.1109/TIT.2004.838106}.'
+
+notes:
+  - 'A kerdock set is a set of symmetric binary matrices with zero diagonal entries with the property that difference of distinct matrices in the set have full rank.'
+  - 'Kerdock codes can be seen as an extended cyclic code over Z4 (quaternary code), where the appended bit is a zero sum check \cite{arXiv:math/0207208}.'
+
+relations:
+  parents:
```

Comment:
2 parents:
bits_into_bits
quaternary_over_z4 with detail mentioning how these nonlinear binary codes are actually linear codes over Z4 via the Gray map
make reed_muller a cousin (kerdock is nonlinear and reedmuller is a child of linear)